onnxruntime/core/providers/cpu/tensor/gather_elements.cc

```diff
@@ -0,0 +1,276 @@
+int64_t compute_base_offset(const std::vector<int64_t>& shape, const TensorPitches& pitches, int64_t skip_axis) {
+  // in this context, rank can never be < 1, so saving checking overhead
+  auto loop_size = static_cast<int64_t>(shape.size()) - 1;
+
+  int64_t base_offset = 0;
+
+  for (int64_t i = 0; i < loop_size; ++i) {
+    if (i != skip_axis)
+      base_offset += (shape[i] * pitches[i]);
+  }
+
+  return base_offset;
+}
+
+// This method computes the number of 'inner_dimension' chunks
+// Example: input = [2, 3]     output = 2
+//          input = [3, 2, 4]  output = 3 * 2 = 6
+//          input  = [2]       output = 1
+int64_t calculate_num_inner_dim(const std::vector<int64_t>& dims) {
+  // in this context, rank can never be < 1, so saving checking overhead
+  int64_t num_inner_dims = 1;
+
+  for (int64_t i = 0; i < static_cast<int64_t>(dims.size()) - 1; ++i) {
+    num_inner_dims *= dims[i];
+  }
+
+  return num_inner_dims;
+}
+
+// This method computes increments over an 'inner_dimension'
+// Example 1: current_dims = [0, 0] tensor_dims = [3, 1], then current_dims = [1, 0]
+//            current_dims = [1, 0] tensor_dims = [2, 1], then current_dims = [2, 0]
+//            current_dims = [2, 0 ] tensor_dims = [ 2, 1 ], then current_dims = [0, 0]
+
+// Example 2: current_dims = [0, 0, 0] tensor_dims = [1, 2, 2], then current_dims = [0, 1, 0]
+//          current_dims = [0, 1, 0] tensor_dims = [1, 2, 2], then current_dims = [0, 0, 0]
+void increment_over_inner_dim(std::vector<int64_t>& current_dims, const std::vector<int64_t>& tensor_dims) {
+  // in this context, rank can never be < 1, so saving checking overhead
+  int64_t rank = static_cast<int64_t>(current_dims.size());
+
+  // 'reset' innermost dimension value
+  current_dims[0] = 0;
+
+  // nothing to increment over
+  if (rank == 1) {
+    return;
+  }
+
+  int64_t current_axis = rank - 2;
+
+  while (current_axis >= 0) {
+    if (++current_dims[current_axis] != tensor_dims[current_axis])
+      return;
+
+    current_dims[current_axis] = 0;
+    --current_axis;
+  }
+}
+
+// parse indices_tensor and along the way validate its shape and contents
+std::vector<int64_t> parse_and_validate_indices_tensor(const Tensor* indices_tensor,
+                                                             int64_t axis, const std::vector<int64_t>& input_shape) {
+  const auto& indices_shape = indices_tensor->Shape().GetDims();
+  int64_t indices_rank = static_cast<int64_t>(indices_shape.size());
+  for (int64_t i = 0; i < indices_rank; ++i) {
+    // for all axes except the axis of interest,
+    // make sure that the corresponding 'indices' shape
+    // value if within bounds of the corresponding 'data' shape
+    if (i != axis) {
+      if (indices_shape[i] < 0 || indices_shape[i] > indices_shape[i])
+        ORT_THROW("GatherElements op: 'indices' shape should have values within bounds of 'data' shape");
+    }
```

Comment:
ORT_THROW("GatherElements op: 'indices' shape should have values within bounds of 'data' shape. Got:", indices_shape);

Side point: usually we throw for things that should not be possible. as this is user input errors most likely can be expected, so returning a failure status would be more consistent with how other operators do it, and you'd get the node name in the error output via the code in the executor.

Reply:
I think after your change, we can get the node name even if it throws....